src/Avalonia.OpenGL/Controls/OpenGlControlBase.cs

```diff
@@ -49,7 +49,10 @@ private void DoCleanup()
 
             _updateQueued = false;
             _visual = null;
-            _resources?.DisposeAsync();
+            if (_resources != null)
+            {
+                await _resources.DisposeAsync();
+            }
             _resources = null;
             _initialization = null;
```

Comment:
I don't think this change is correct. 
"_resources" will be hold in the disposing async task, while "_resources" field is still present and not null.
With the previous code, there weren't any problems.